soh/soh/Enhancements/debugger/actorViewer.cpp

```diff
@@ -0,0 +1,1261 @@
+            if (fetch != NULL) {
+                display = *fetch;
+                category = fetch->category;
+                PopulateActorDropdown(category, list);
+                rm = TARGET;
+            }
+        }
+        InsertHelpHoverText("Grabs actor with target arrow above it. You might need C-Up for enemies");
+        if (ImGui::Button("Fetch from Held")) {
+            Player* player = GET_PLAYER(gGlobalCtx);
+            fetch = player->heldActor;
+            if (fetch != NULL) {
+                display = *fetch;
+                category = fetch->category;
+                PopulateActorDropdown(category, list);
+                rm = HELD;
+            }
+        }
+        InsertHelpHoverText("Grabs actor that Link is holding");
+        if (ImGui::Button("Fetch from Interaction")) {
+            Player* player = GET_PLAYER(gGlobalCtx);
+            fetch = player->interactRangeActor;
+            if (fetch != NULL) {
+                display = *fetch;
+                category = fetch->category;
+                PopulateActorDropdown(category, list);
+                rm = INTERACT;
+            }
+        }
+        InsertHelpHoverText("Grabs actor from \"interaction range\"");
+
+        ImGui::TreePop();
+    }
+    
+    if (ImGui::TreeNode("New...")) {
+        ImGui::PushItemWidth(ImGui::GetFontSize() * 10);
+
+        ImGui::InputScalar("ID", ImGuiDataType_S16, &newActor.id, &one);
+        ImGui::InputScalar("params", ImGuiDataType_S16, &newActor.params, &one);
+
+        ImGui::PushItemWidth(ImGui::GetFontSize() * 6);
+
+        DrawGroupWithBorder([&]() {
+            ImGui::Text("New Actor Position");
+            ImGui::InputScalar("posX", ImGuiDataType_Float, &newActor.pos.x);
+            ImGui::SameLine();
+            ImGui::InputScalar("posY", ImGuiDataType_Float, &newActor.pos.y);
+            ImGui::SameLine();
+            ImGui::InputScalar("posZ", ImGuiDataType_Float, &newActor.pos.z);
+        });
+
+        DrawGroupWithBorder([&]() {
+            ImGui::Text("New Actor Rotation");
+            ImGui::InputScalar("rotX", ImGuiDataType_S16, &newActor.rot.x);
+            ImGui::SameLine();
+            ImGui::InputScalar("rotY", ImGuiDataType_S16, &newActor.rot.y);
+            ImGui::SameLine();
+            ImGui::InputScalar("rotZ", ImGuiDataType_S16, &newActor.rot.z);
+        });
+
+        if (ImGui::Button("Fetch from Link")) {
+            Player* player = GET_PLAYER(gGlobalCtx);
+            Vec3f newPos = player->actor.world.pos;
+            Vec3s newRot = player->actor.world.rot;
+            newActor.pos = newPos;
+            newActor.rot = newRot;
+        }
+
+        if (ImGui::Button("Spawn")) {
+            Actor_Spawn(&gGlobalCtx->actorCtx, gGlobalCtx, newActor.id, newActor.pos.x, newActor.pos.y, newActor.pos.z,
+                        newActor.rot.x, newActor.rot.y, newActor.rot.z, newActor.params);
+        }
+
+        if (ImGui::Button("Spawn as Child")) {
```

Comment:
IMO this should take an actor to make a parent and not the first in the list.

Reply:
Any suggestions on how I could go about that? I was using [this snippet from gz's debug section](https://github.com/glankk/gz/blob/master/src/gz/gz_debug.c#L504) and it looks like they also take the head of the respective `ActorListEntry`.